Makefile

```
@@ -1127,3 +1135,88 @@ $(GOLANGCI_LINT): .github/workflows/golangci-lint.yml # Download golangci-lint u
 	hack/ensure-golangci-lint.sh \
 		-b $(TOOLS_BIN_DIR) \
 		$(shell cat .github/workflows/golangci-lint.yml | grep [[:space:]]version | sed 's/.*version: //')
+
+$(GINKGO): # Build ginkgo from tools folder.
+	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) $(GINKGO_PKG) $(GINKGO_BIN) $(GINGKO_VER)
+
+##@ end-to-end
+
+## --------------------------------------
+## Templates
+## --------------------------------------
+
+DOCKER_TEMPLATES := test/e2e/data/infrastructure-docker
+
+.PHONY: cluster-templates
+cluster-templates: $(KUSTOMIZE) cluster-templates-v1alpha3 cluster-templates-v1alpha4 cluster-templates-v1.2 cluster-templates-v1beta1 ## Generate cluster templates for all versions
+
+cluster-templates-v1alpha3: $(KUSTOMIZE) ## Generate cluster templates for v1alpha3
```

```suggestion
.PHONY: generate-e2e-templates-v1alpha3
cluster-templates-v1alpha3: $(KUSTOMIZE)
```
(removing the comment so this target is not visible in the menu; basically we can keep sub-targets for code organization, but no need to surface them to the user given that generate-e2e-templates takes only few seconds)